boot/bootstate20.go

```diff
@@ -219,6 +222,38 @@ func (u20 *bootStateUpdate20) commit() error {
 		expectReseal = resealExpectedByModeenvChange(u20.writeModeenv, u20.modeenv)
 	}
 
+	autoRepair, err := IsUnlockedWithRecoveryKey()
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+	} else if autoRepair {
+		// Remove cache to force resealing
+		for _, cache := range []string{
+			filepath.Join(dirs.SnapFDEDir, "boot-chains"),
+			filepath.Join(dirs.SnapFDEDir, "recovery-boot-chains"),
```

Comment:
This will force resealing.

Reply:
this works has a quick things but we probably want a todo about a force flag to the helper below

Reply:
I will change that before it is ready for merge. That was to make the simple patch for the first pass.